services/workflows-service/src/workflow/workflow.controller.internal.ts

```diff
@@ -133,19 +139,44 @@ export class WorkflowControllerInternal {
   }
 
   @common.Post('/:id/event')
-  @swagger.ApiOkResponse()
-  @common.HttpCode(200)
-  @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  @ApiResponse({
+    status: 400,
+    description: 'Validation error',
+    schema: Type.Object({
+      errorCode: Type.Literal('BadRequest'),
+      message: Type.String(),
+      statusCode: Type.Literal(400),
+      timestamp: Type.String({
+        format: 'date-time',
+      }),
+      path: Type.String(),
+      errors: Type.Array(Type.Object({ message: Type.String(), path: Type.String() })),
+    }),
+  })
```

Comment:
We are going to use it in many more places, right? If so, extracting it into a reusable decorator makes sense. WDYT about `@ApiValidationErrorResponse()`?